src/javascript/app/pages/trade/purchase/purchase.jsx

```diff
@@ -84,110 +85,101 @@ const Purchase = () => {
-                                <CaptionText centered >{parseData(data.topComment)}</CaptionText>
-                            </div>
-                        </div>
-                        <div className='purchase-box'>
-                            <div className='purchase-header'>
-                                <div className='purchase-icon-box'>
-                                    <div className={`contract_heading ${data?.bottomContractType}`} />
-                                    <Text size='md' centered bold> {data?.bottomDisplayText}</Text>
-                                </div>
-                                <div className='purchase-body-box'>
-                                    <div className='purchase-amount-box'>
-                                        <div className='purchase-amount-info-box'>
-                                            <Text size='md'>{localize('Stake')}:</Text>
-                                            <Text size='md' className={data.bottomAmountClassname} bold>{data?.bottomAmount} {displayCurrency(data?.bottomAmount)}</Text>
+                        {purchase_positions.map(position => {
+                            const contract_type  = data?.[`${position}_contract_type`];
+                            const display_text  = data?.[`${position}_display_text`];
+                            const amount  = data?.[`${position}_amount`];
+                            const amount_classname  = data?.[`${position}_amount_classname`];
+                            const payout_amount_classname  = data?.[`${position}_payout_amount_classname`];
+                            const payout_amount  = data?.[`${position}_payout_amount`];
+                            const purchase_disabled  = data?.[`${position}_purchase_disabled`];
+                            const description  = data?.[`${position}_description`];
+                            const comment  = data?.[`${position}_comment`];
+                        
+                            return (
+                                <Fragment key={`purchase-action-${position}`}>
+                                    <div className='purchase-box'>
+                                        <div className='purchase-header'>
+                                            <div className='purchase-icon-box'>
+                                                <div className={`contract_heading ${contract_type}`} />
+                                                <Text size='md' centered bold> {display_text}</Text>
+                                            </div>
+                                          
+                                            <div className='purchase-body-box'>
+                                                <div className='purchase-amount-box'>
+                                                    <div className='purchase-amount-info-box'>
+                                                        <Text size='md'>{localize('Stake')}:</Text>
```

Comment:
```suggestion
                                                        <Text>{localize('Stake')}:</Text>
```
nit-pick: not sure if it's in scope of this PR but `size='md'` can be removed from all Text's as it's a default value :)